src/Xray/XrayScanClient.ts

```diff
@@ -11,29 +12,41 @@ export class XrayScanClient {
 
     public async graph(
         model: IGraphRequestModel,
+        progress: XrayScanProgress,
         checkCanceled: () => void,
         projectKey: string | undefined,
         sleepIntervalMilliseconds: number = XrayScanClient.SLEEP_INTERVAL_MILLISECONDS
     ): Promise<IGraphResponse> {
-        if (!model) {
-            return {} as IGraphResponse;
-        }
+        try {
+            if (!model) {
+                return {} as IGraphResponse;
+            }
 
-        const projectProvided: boolean = projectKey !== undefined && projectKey.length > 0;
-        this.logger.debug('Sending POST scan/graph request...');
-        const requestParams: IRequestParams = {
-            url: XrayScanClient.scanGraphEndpoint + (projectProvided ? `?project=${projectKey}` : ''),
-            method: 'POST',
-            data: model,
-        };
-        this.logger.debug('data: ' + JSON.stringify(model));
-        checkCanceled();
-        const response: IGraphResponse = await this.httpClient.doAuthRequest(requestParams);
-        return this.getScanGraphResults(response.scan_id, checkCanceled, !projectProvided, sleepIntervalMilliseconds);
+            const projectProvided: boolean = projectKey !== undefined && projectKey.length > 0;
+            this.logger.debug('Sending POST scan/graph request...');
+            const requestParams: IRequestParams = {
+                url: XrayScanClient.scanGraphEndpoint + (projectProvided ? `?project=${projectKey}` : ''),
+                method: 'POST',
+                data: model,
+            };
+            this.logger.debug('data: ' + JSON.stringify(model));
+            checkCanceled();
+            const response: IGraphResponse = await this.httpClient.doAuthRequest(requestParams);
+            return await this.getScanGraphResults(
+                response.scan_id,
+                progress,
+                checkCanceled,
+                !projectProvided,
+                sleepIntervalMilliseconds
+            );
+        } finally {
+            progress.setPercentage(100);
+        }
     }
 
     private async getScanGraphResults(
         scanId: string,
+        progress: XrayScanProgress,
```

Comment:
This code flow expects either 200 or 202 responses and acts differently. I suggest we add some inline documentation here, explaining the responses, the progress reporting, the max attempts and so on. I believe this will make the code easier to read.